crates/topos-p2p/src/network.rs

```diff
@@ -39,8 +39,8 @@ const TWO_HOURS: Duration = Duration::from_secs(60 * 60 * 2);
 pub struct NetworkBuilder<'a> {
     discovery_protocol: Option<&'static str>,
     peer_key: Option<Keypair>,
-    listen_addr: Option<Multiaddr>,
-    exposed_addresses: Option<Multiaddr>,
+    listen_addresses: Option<Vec<Multiaddr>>,
+    advertised_addresses: Option<Vec<Multiaddr>>,
```

Comment:
To be consistent I think we should use `listening_addresses` and `advertised_addresses` or `listen_addresses` and `advertise_addresses`. First one sounds better to me